src/wp-includes/media.php

```diff
@@ -5640,9 +5693,11 @@ function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
 		return $loading_attrs;
 	}
 
+	$maybe_size_less_attrs = null;
```

Comment:
See my other feedback, we can apply the `decoding` attribute logic right here, before the early return for missing dimension attributes. We only need to then move the context-specific early return above this.
```suggestion
	/*
	 * Add `decoding` with a value of "async" for every image unless it has a
	 * conflicting `decoding` attribute already present.
	 */
	if ( 'img' === $tag_name ) {
		if ( isset( $attr['decoding'] ) ) {
			if ( 'async' === $attr['decoding'] ) {
				$loading_attrs['decoding'] = 'async';
			}
		} else {
			$loading_attrs['decoding'] = 'async';
		}
	}
```